product/views.py

```diff
@@ -1,3 +1,67 @@
-from django.shortcuts import render
+# 이 상품 어때요?
+import random
 
-# Create your views here.
+from django.views import View
+from django.http  import JsonResponse
+
+from .models import Category, SubCategory, Product, DiscountRate
+
+
+# 카테고리 무작위로 12개, 
+# 그리고 카테고리별 sub 카테고리는 각 1개, 
+# 그 sub 카테고리의 랜덤 상품 1개
+# 5분에 한 번씩 상품 추천 리스트가 변하도록 구현해보자 (random.seed()???)
+class RecommendView(View):
+    def get(self, request):
+        try:
+            categories        = Category.objects.all()
+            random_categories = random.sample(list(categories), 8) # 랜덤으로 뽑을 개수. 나중에는 이 개수를 12개로 늘린다.
+            
+            # 랜덤 카테고리를 기준으로 랜덤 서브 카테고리를 1개씩 추출
+            random_sub_category_list = []
+
+            for category in random_categories:
+                sub_categories      = category.subcategory_set.all()
+                random_sub_category = random.choice(sub_categories)
+
+                random_sub_category_list.append(random_sub_category)
```

Comment:
```suggestion
            random_sub_category_list = []

            for category in random_categories:
                sub_categories      = category.subcategory_set.all()
                random_sub_category = random.choice(sub_categories)

                random_sub_category_list.append(random_sub_category)
```
리스트 컴프리헨션으로 변경해봅시다~!

Reply:
네 알겠습니다~!